server/models/services/GithubWorkflowService.js

```diff
@@ -0,0 +1,157 @@
+// >   app. For example, you can get an installation ID with the `GET
+// >   /users/{username}/installation`, `GET
+// >   /repos/{owner}/{repo}/installation`, `GET /orgs/{org}/installation`, or
+// >   `GET /app/installations endpoints`. For more information, see
+// >   "[GitHub Apps](https://docs.github.com/en/rest/apps/apps)".
+//
+// https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/authenticating-as-a-github-app-installation
+const GITHUB_APP_INSTALLATION_ID = '42217598';
+
+// > The time that the JWT was created. To protect against clock drift, we
+// > recommend that you set this 60 seconds in the past and ensure that your
+// > server's date and time is set accurately (for example, by using the Network
+// > Time Protocol).
+//
+// https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/generating-a-json-web-token-jwt-for-a-github-app
+const calculateIssuedAt = () => Math.round(Date.now() / 1000) - ONE_MINUTE;
+
+const createJWT = (payload, privateKey, algorithm) => {
+    return new Promise((resolve, reject) => {
+        jwt.sign(payload, privateKey, { algorithm }, (err, token) => {
+            token ? resolve(token) : reject(err);
+        });
+    });
+};
+
+// > The expiration time of the JWT, after which it can't be used to request an
+// > installation token. The time must be no more than 10 minutes into the
+// > future.
+//
+// https://docs.github.com/en/apps/creating-github-apps/authenticating-with-a-github-app/generating-a-json-web-token-jwt-for-a-github-app
+const calculateExpiresAt = () => Math.round(Date.now() / 1000) + 9 * ONE_MINUTE;
+
+const fetchInstallationAccessToken = async (jsonWebToken, installationID) => {
+    const response = await fetch(
+        `https://api.github.com/app/installations/${installationID}/access_tokens`,
+        {
+            method: 'POST',
+            headers: {
+                Accept: 'application/vnd.github+json',
+                Authorization: `Bearer ${jsonWebToken}`,
+                'X-GitHub-Api-Version': '2022-11-28'
+            }
+        }
+    );
+
+    if (!response.ok) {
+        const message = await (() => {
+            try {
+                return response.text();
+            } catch (_) {
+                return 'Unable to retrieve installation access token';
+            }
+        })();
+
+        throw new Error(message);
+    }
+
+    return (await response.json()).token;
+};
+
+const createGithubWorkflow = async ({
+    job,
+    directory = 'alert',
+    gitSha = 'master'
+}) => {
+    const payload = {
+        iat: calculateIssuedAt(),
+        exp: calculateExpiresAt(),
+        iss: GITHUB_APP_ID
+    };
+    const jsonWebToken = await createJWT(payload, privateKey, ALGORITHM);
+    const accessToken = await fetchInstallationAccessToken(
+        jsonWebToken,
+        GITHUB_APP_INSTALLATION_ID
+    );
+    const inputs = {
+        nonce: job.id,
```

Comment:
Does the system use this?

Reply:
not actively, it does send it back but it fairly "unneeded" in the current setup

Reply:
Let's 86 it, then